thorlcr/slave/slave.cpp

```diff
@@ -173,12 +168,13 @@ void ProcessSlaveActivity::serializeStats(MemoryBuffer &mb)
 #ifdef TIME_ACTIVITIES
     if (timeActivities)
     {
-        SpinBlock b(cycleLock);
-        if (lastCycles)
+        unsigned __int64 curCycles = lastCycles;
+        if (curCycles)
         {
             unsigned __int64 nowCycles = get_cycles_now();
-            totalCycles.totalCycles += nowCycles-lastCycles;
-            lastCycles = nowCycles; // time accounted for
+            //Update lastCycles to the current number of cycles - unless it has been set to 0 in the meantime
+            if (lastCycles.compare_exchange_strong(curCycles, nowCycles))
```

Comment:
Do we really need seq_cst memory order ?

Reply:
no, I will reduce it.